user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/UserManagementApp.java

```diff
@@ -0,0 +1,92 @@
+
+import tech.jhipster.config.DefaultProfileUtil;
+import tech.jhipster.config.JHipsterConstants;
+
+@EntityScan("de.tum.in.www1.artemis")
+@SpringBootApplication(scanBasePackages = { "de.tum.in.www1.artemis" })
+@EnableConfigurationProperties(LiquibaseProperties.class)
+public class UserManagementApp {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserManagementApp.class);
+
+    private final Environment env;
+
+    public UserManagementApp(Environment env) {
+        this.env = env;
+    }
+
+    /**
+     * Main method, used to run the application.
+     *
+     * @param args the command line arguments.
+     */
+    public static void main(String[] args) {
+        SpringApplication app = new SpringApplication(UserManagementApp.class);
+        DefaultProfileUtil.addDefaultProfile(app);
+        Environment env = app.run(args).getEnvironment();
+        logApplicationStartup(env);
+    }
+
+    private static void logApplicationStartup(Environment env) {
+        String protocol = Optional.ofNullable(env.getProperty("server.ssl.key-store")).map(key -> "https").orElse("http");
+        String serverPort = env.getProperty("server.port");
+        String contextPath = Optional.ofNullable(env.getProperty("server.servlet.context-path")).filter(StringUtils::isNotBlank).orElse("/");
+        String hostAddress = "localhost";
+        try {
+            hostAddress = InetAddress.getLocalHost().getHostAddress();
+        }
+        catch (UnknownHostException e) {
+            LOGGER.warn("The host name could not be determined, using `localhost` as fallback");
+        }
+        LOGGER.info(
+                "\n----------------------------------------------------------\n\t" + "Application '{}' is running! Access URLs:\n\t" + "Local: \t\t{}://localhost:{}{}\n\t"
+                        + "External: \t{}://{}:{}{}\n\t" + "Profile(s): \t{}\n----------------------------------------------------------",
+                env.getProperty("spring.application.name"), protocol, serverPort, contextPath, protocol, hostAddress, serverPort, contextPath, env.getActiveProfiles());
+
+        String configServerStatus = env.getProperty("configserver.status");
+        if (configServerStatus == null) {
+            configServerStatus = "Not found or not setup for this application";
+        }
+        LOGGER.info("\n----------------------------------------------------------\n\t" + "Config Server: \t{}\n----------------------------------------------------------",
+                configServerStatus);
+    }
+
+    /**
+     * Initializes userManagement.
+     * <p>
+     * Spring profiles can be configured with a program argument --spring.profiles.active=your-active-profile
+     * <p>
+     * You can find more information on how profiles work with JHipster on <a href="https://www.jhipster.tech/profiles/">https://www.jhipster.tech/profiles/</a>.
+     */
+    @PostConstruct
+    public void initApplication() {
```

Comment:
Not sure if you think it's worth thinking about this now but this setup seems to be another cause of code duplication to me.
Especially with more microservices in the future. Do you know if Spring provides a solution for that? This could, however, also be tackled in a follow-up since it relates to both, gateway and user management.

Reply:
I think it's normal to have such duplicates related to setting up the project as we are creating a new project. I will think about it and may include it in a follow-up.